pkg/util/yaml/decoder.go

```diff
@@ -52,10 +53,16 @@ type YAMLToJSONDecoder struct {
 // stream in chunks by converting each document (as defined by
 // the YAML spec) into its own chunk, converting it to JSON via
 // yaml.YAMLToJSON, and then passing it to json.Decoder.
-func NewYAMLToJSONDecoder(r io.Reader) *YAMLToJSONDecoder {
+func NewYAMLToJSONDecoder(r io.Reader, bytes []byte, isStdin bool) *YAMLToJSONDecoder {
 	reader := bufio.NewReader(r)
-	return &YAMLToJSONDecoder{
-		reader: NewYAMLReader(reader),
+	if isStdin && sessionIsTTY() {
+		return &YAMLToJSONDecoder{
+			reader: NewYAMLByteReader(bytes),
```

Comment:
What I could do is the following. If there are more bytes than 4096, the error returned will be `ErrBufferFull`. 
- If this error is there, I'll keep the current behaviour of the code (with the issue where you have to press `Ctrl + D` twice)
- If this error is not there, I can use my current solution.

This way the problem is solved for most cases and the code still supports bigger files. 

I was looking for the example files like this one and they are usually ~1100 bytes: https://github.com/kubernetes/kubernetes/blob/master/examples/guestbook/legacy/frontend-controller.yaml

A user would have to enter the equivalent of ~3 yaml files in the stdin stream to reach a `ErrBufferFull` error.